internal/sink/matcher.go

```diff
@@ -47,9 +47,24 @@ type CommonWebhook struct {
 	logger *zap.Logger
 }
 
+func NewMatcher(l Listener, logger *zap.Logger) (matcher Matcher, version int, err error) {
```

Comment:
this is more of an artifact from the past, let's take a note to review how loggers are used and passed around in caduceus in general